### PR TITLE
Added a few commands, fixed id -> cid

### PR DESCRIPTION
**Pull request by SubstituteCS**

### CODE REVIEW

**Comment by nikeee** [comment]
Nice, thanks!
Why did you remove the ConfigureAwait stuff?

**Comment by SubstituteCS** [comment]
Because I don't have resharper.

**Comment by nikeee** [comment]
It's not a feature of ReSharper - which I don't have either. The ConfigureAwait is there for performance reasons.
Could you revert the removal of the ConfigureAwait? I'd like to merge this PR after that.

**Comment by SubstituteCS** [comment]
Reverted.  For some reason my copy of Visual Studio was treating it as a piece of resharper, but it seems to have resolved on it's own.

**Comment by Splamy** [comment]
Now there are some duplicate semicolon lines, and a few configureAwaits are still missing

**Comment by SubstituteCS** [comment]
Sorry, did it over lunch in like 10 minutes.  Should be fixed for _real_ now.

File: src/TeamSpeak3QueryApi/Specialized/TeamSpeakClient.cs [hunk 1]
```diff
@@ -385,8 +385,49 @@ public Task DeleteChannel(int channelId, bool force)
...
+            if (channel.PhoneticName != null) { UpdateParameters.Add(new Parameter("channel_name_phonetic", channel.PhoneticName)); }
+            if (channel.IconId != null) { UpdateParameters.Add(new Parameter("channel_icon_id", (int)channel.IconId)); }
+            if (channel.IsCodecUnencrypted != null) { UpdateParameters.Add(new Parameter("channel_codec_is_unencrypted", channel.IsCodecUnencrypted)); }
+            if (channel.ParentChannelId != null) { UpdateParameters.Add(new Parameter("channel_cpid", channel.ParentChannelId)); }
+            return Client.Send("channeledit", UpdateParameters.ToArray());
+        }
 
         #endregion
+        #region ChannelAddPerm
+        public Task ChannelAddPerm(int ChannelId, string PermsId, int PermValue)
```
Comment (nikeee) [inline src/TeamSpeak3QueryApi/Specialized/TeamSpeakClient.cs]: Could you fix the casing of the parameters making them camelCase?

File: src/TeamSpeak3QueryApi/Specialized/TeamSpeakClient.cs [hunk 2]
```diff
@@ -478,8 +519,25 @@ public Task PokeClient(int clientId, string message)
 
         #endregion
 
+        #region ChangeNickName
+        public Task ChangeNickName(string NickName)
```
Comment (nikeee) [inline src/TeamSpeak3QueryApi/Specialized/TeamSpeakClient.cs]: Casing again

File: src/TeamSpeak3QueryApi/Specialized/TeamSpeakClient.cs [hunk 3]
```diff
@@ -478,8 +519,25 @@ public Task PokeClient(int clientId, string message)
...
+        {
+            return ChangeNickName(NickName, null);
+        }
+
+        public Task ChangeNickName(string NickName, WhoAmI WhoAmI)
```
Comment (nikeee) [inline src/TeamSpeak3QueryApi/Specialized/TeamSpeakClient.cs]: See above

File: src/TeamSpeak3QueryApi/TeamSpeak3QueryApi.csproj [hunk 4]
```diff
@@ -103,9 +104,6 @@
   <ItemGroup>
     <None Include="packages.config" />
   </ItemGroup>
-  <ItemGroup>
-    <Analyzer Include="..\packages\ConfigureAwaitChecker.Analyzer.1.0.0-beta4\analyzers\dotnet\cs\ConfigureAwaitChecker.dll" />
-  </ItemGroup>
```
Comment (nikeee) [inline src/TeamSpeak3QueryApi/TeamSpeak3QueryApi.csproj]: This analyzer is used to see if there could be some ConfigureAwaits put somewhere. We need this as part of the build. Could you revert this removal?

File: src/TeamSpeak3QueryApi/Specialized/TeamSpeakClient.cs [hunk 5]
```diff
@@ -385,8 +385,49 @@ public Task DeleteChannel(int channelId, bool force)
         }
 
         #endregion
+        #region EditChannel
+
+        public Task EditChannel(int channelid, EditChannelInfo channel)
+        {
+            List<Parameter> UpdateParameters = new List<Parameter>();
+
+            UpdateParameters.Add(new Parameter("cid", channelid));
```
Comment (nikeee) [inline src/TeamSpeak3QueryApi/Specialized/TeamSpeakClient.cs]: See above

**Pull request merged**